### PR TITLE
Fix wiggling on edges of floors/ceilings

### PR DESCRIPTION
**Pull request by facespkz**
Ports Crispy's wiggle fix code to prevent distorted floors and ceilings.

### CODE REVIEW

**Review by rfomin — APPROVED** [review]
Thank you for this! Looks good to me, but it seems we'll have to wait a few days for @fabiangreffrath's review.

**Comment by fabiangreffrath** [comment]
I remember I once had this in a very early branch but rejected it, because I found the port doesn't really need it (because of its limited resolution). What it really benefited from was the necessary upgrade of renderer variables to 32-bit integers, though, which has already been implemented independent if the actual wiggle fix.

File: Source/r_segs.c [hunk 1]
```diff
@@ -83,6 +83,101 @@ static int64_t  bottomfrac; // [FG] 64-bit integer math
 static fixed_t  bottomstep;
 static int    *maskedtexturecol; // [FG] 32-bit integer math
 
+// WiggleFix: add this code block near the top of r_segs.c
```
Comment (fabiangreffrath) [inline Source/r_segs.c:86]: Could you move this down a bit, right before `R_RenderSegLoop()` so you could use the `HEIGHTBITS` and `HEIGHTUNIT` macros for initialization of the `heightbits` and `heightunit` variables?

File: Source/r_main.c [hunk 2]
```diff
@@ -224,6 +224,9 @@ angle_t R_PointToAngleCrispy(fixed_t x, fixed_t y)
     0;
 }
 
+// WiggleFix: move R_ScaleFromGlobalAngle to r_segs.c,
+// above R_StoreWallRange
+#if 0
```
Comment (fabiangreffrath) [inline Source/r_main.c:229]: Please remove this code, instead of commenting it out, but leave the comment in place.

**Review by fabiangreffrath — CHANGES_REQUESTED** [review]

File: Source/r_segs.c [hunk 3]
```diff
@@ -83,6 +83,7 @@ static int64_t  bottomfrac; // [FG] 64-bit integer math
 static fixed_t  bottomstep;
 static int    *maskedtexturecol; // [FG] 32-bit integer math
 
+
```
Comment (fabiangreffrath) [inline Source/r_segs.c:86]: Excess blank line.

Comment (facespkz) [reply Source/r_segs.c:86]: oops

**Pull request merged**